packages/ember-application/tests/system/instance_initializers_test.js

```diff
@@ -1,410 +1,386 @@
-  equal(firstInitializerRunCount, 1, 'first initializer was run when subclass created');
-  equal(secondInitializerRunCount, 1, 'second initializers was run when subclass created');
-  run(function() {
-    firstApp.destroy();
-    secondApp.destroy();
-  });
-});
 
-QUnit.test('initializers are per-app', function() {
-  expect(2);
+    this.runTask(() => this.createApplication({}, FirstApp));
 
-  let FirstApp = Application.extend();
+    equal(firstInitializerRunCount, 1, 'first initializer only was run when base class created');
+    equal(secondInitializerRunCount, 0, 'first initializer only was run when base class created');
 
-  FirstApp.instanceInitializer({
-    name: 'abc',
-    initialize(app) {}
-  });
+    firstInitializerRunCount = 0;
+    this.runTask(() => this.createSecondApplication({}, SecondApp));
 
-  expectAssertion(function() {
+    equal(firstInitializerRunCount, 1, 'first initializer was run when subclass created');
+    equal(secondInitializerRunCount, 1, 'second initializers was run when subclass created');
+  }
+
+  [`@test initializers are per-app`](assert) {
+    assert.expect(2);
+
+    let FirstApp = Application.extend();
     FirstApp.instanceInitializer({
       name: 'abc',
       initialize(app) {}
     });
-  });
 
-  let SecondApp = Application.extend();
-  SecondApp.instanceInitializer({
-    name: 'abc',
-    initialize(app) {}
-  });
+    expectAssertion(function() {
+      FirstApp.instanceInitializer({
+        name: 'abc',
+        initialize(app) {}
+      });
+    });
 
-  ok(true, 'Two apps can have initializers named the same.');
-});
+    this.runTask(() => this.createApplication({}, FirstApp));
 
-QUnit.test('initializers are run before ready hook', function() {
-  expect(2);
+    let SecondApp = Application.extend();
+    SecondApp.instanceInitializer({
+      name: 'abc',
+      initialize(app) {}
+    });
 
-  let readyWasCalled = false;
+    this.runTask(() => this.createSecondApplication({}, SecondApp));
 
-  let MyApplication = Application.extend({
-    ready() {
-      ok(true, 'ready is called');
-      readyWasCalled = true;
-    }
-  });
+    assert.ok(true, 'Two apps can have initializers named the same.');
+  }
 
-  MyApplication.instanceInitializer({
-    name: 'initializer',
-    initialize() {
-      ok(!readyWasCalled, 'ready is not yet called');
-    }
-  });
+  [`@test initializers are run before ready hook`](assert) {
+    expect(2);
 
-  run(function() {
-    app = MyApplication.create({
-      router: false,
-      rootElement: '#qunit-fixture'
+    let MyApplication = Application.extend();
+    let readyWasCalled = false;
+
+    MyApplication.instanceInitializer({
+      name: 'initializer',
+      initialize() {
+        assert.ok(!readyWasCalled, 'ready is not yet called');
+      }
     });
-  });
-});
 
-QUnit.test('initializers are executed in their own context', function() {
-  expect(1);
+    this.runTask(() => {
+      this.createApplication({
+        ready() {
```

Comment:
Can this `ready() {` stay on the extended object instead of being moved to the `create` options?